api/src/main/java/me/lokka30/treasury/api/economy/transaction/EconomyTransaction.java

```diff
@@ -259,6 +270,17 @@ public Builder withTransactionAmount(BigDecimal transactionAmount) {
             return this;
         }
 
+        /**
+         * Specify the importance this transaction should have.
+         *
+         * @param importance importance of the transaction
+         * @return this insatance for chaining
+         */
+        public Builder withImportance(EconomyTransactionImportance importance) {
```

Comment:
Parameter should likely be NotNull (as should the other builder methods), and if you're really feeling it all the builder methods could also have `@Contract("_ -> this")`